modules/tensor_mechanics/src/utils/ElasticityTensorTools.C

```diff
@@ -21,11 +21,42 @@ elasticJacobian(const RankFourTensor & r4t,
 {
   // d(stress_ij*d(test)/dx_j)/du_k = d(C_ijmn*du_m/dx_n dtest/dx_j)/du_k (which is nonzero for m ==
   // k)
-  Real sum = 0.0;
-  for (unsigned int j = 0; j < LIBMESH_DIM; ++j)
-    for (unsigned int l = 0; l < LIBMESH_DIM; ++l)
-      sum += r4t(i, j, k, l) * grad_phi(l) * grad_test(j);
-  return sum;
+
+  Real gt0 = grad_test(0);
+  Real gt1 = grad_test(1);
+  Real gt2 = grad_test(2);
+  Real gp0 = grad_phi(0);
+  Real gp1 = grad_phi(1);
+  Real gp2 = grad_phi(2);
```

Comment:
again, all those should be `const`. If only for the reader to know that you are basically defining shorr-hands.